### ISSUE TITLE
Check memory read

### ISSUE THREAD

**Issue by zayip** [brevent/genuine#2]
Loved this professional codes, tq.
Btw I have some questions, there's some app call gâmegu@rdìan, it's able to do memory scan and read on the app. Tested g@meguårdi@n on brevent app, it works. Is there any way to detect memory read or block it using genuine?

**Comment by liudongmiao** [comment]
Sorry, but I don't get you.

And if devices have rooted, it's ok for root to read the entire memory.
So, scan for what? Do you means for scan some special codes or strings?

And please give me some poc for tampered brevent.
You can emailed my via brevent@jianyu.io

**Comment by zayip** [comment]
Yes, GG app (apk available online) able to read sensitive strings/values that doesn't show on app UI, you also can modify the strings using that app too.

**Comment by liudongmiao** [comment]
I don't think apps can bypass Genuine's signature check.

It means, Genuine don't anti root, only anti hijack or signature-tamper.

**Comment by zayip** [comment]
Ok understand sir, after googling I found need to use anti pttace or anti maps read to prevent this.

**Comment by liudongmiao** [comment]
Anti root / ptrace / debug is not part of Genuine.

And there is no way to anti read maps, root user can always read it.

However, Genuine implement a way to anti fake-maps, like other app hijack open or similiar api to return a fake maps.

For sensitive content, I suggest don't put it in memory, just encrypt, and decrypt to use, then encrypt to restore in memory.

**Comment by zayip** [comment]
Thank you for your information, you are kind. 
Oh... what happen if I remove 'extractNativeLibs' what will be broken? I think working fine on android6 from your sample even I removed

**Comment by liudongmiao** [comment]
Default value for `extractNativeLibs` is true.

Add `extractNativeLibs` in library make a warning if you use `extractNativeLibs` with `false` on Android 6.0 devices.

**Comment by febryanasaperdana** [comment]
> Anti root / ptrace / debug is not part of Genuine.
> 
> And there is no way to anti read maps, root user can always read it.
> 
> However, Genuine implement a way to anti fake-maps, like other app hijack open or similiar api to return a fake maps.
> 
> For sensitive content, I suggest don't put it in memory, just encrypt, and decrypt to use, then encrypt to restore in memory.

What do you mean by anti-fake maps? It means that we fill our /proc/self/maps with fake content instead of the real one?
